e2e/security/security.go

```diff
@@ -0,0 +1,218 @@
+		// capabilities
+		{
+			name:       "capabilities_keep_true",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--keep-privs"},
+			expectExit: 255,
+		},
+		{
+			name:       "capabilities_keep-false",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			expectExit: 2,
+		},
+		{
+			name:       "capabilities_drop",
+			image:      c.env.ImagePath,
+			action:     "exec",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--drop-caps", "CAP_NET_RAW"},
+			expectExit: 2,
+		},
+	}
+
+	for _, tt := range tests {
+		optArgs := []string{}
+		optArgs = append(optArgs, tt.opts...)
+		optArgs = append(optArgs, c.env.ImagePath)
+		optArgs = append(optArgs, tt.argv...)
+
+		c.env.RunSingularity(
+			t,
+			e2e.AsSubtest(tt.name),
+			e2e.WithCommand("exec"),
+			e2e.WithArgs(optArgs...),
+			e2e.ExpectExit(tt.expectExit),
+		)
+
+	}
+}
+
+// testSecurityPriv tests security flag fuctionality for singularity exec with elevated privileges
+func (c *ctx) testSecurityPriv(t *testing.T) {
+	tests := []struct {
+		name       string
+		argv       []string
+		opts       []string
+		expectExit int
+	}{
+		// taget UID/GID
+		{
+			name:       "Set_uid",
+			argv:       []string{"id", "-u", "|", "grep", "99"},
+			opts:       []string{"--security", "uid:99"},
+			expectExit: 1,
+		},
+		{
+			name:       "Set_gid",
+			argv:       []string{"id", "-g", "|", "grep", "99"},
+			opts:       []string{"--security", "gid:99"},
+			expectExit: 1,
+		},
+		// seccomp from json file
+		{
+			name:       "SecComp_BlackList",
+			argv:       []string{"mkdir", "/tmp/foo"},
+			opts:       []string{"--security", "seccomp:./testdata/seccomp-profile.json"},
+			expectExit: 159,
+		},
+		{
+			name:       "SecComp_true",
+			argv:       []string{"true"},
+			opts:       []string{"--security", "seccomp:./testdata/seccomp-profile.json"},
+			expectExit: 0,
+		},
+		// capabilities
+		{
+			name:       "capabilities_keep",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--keep-privs"},
+			expectExit: 0,
+		},
+		{
+			name:       "capabilities_drop",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--drop-caps", "CAP_NET_RAW"},
+			expectExit: 2,
+		},
+	}
+
+	for _, tt := range tests {
+		optArgs := []string{}
+		optArgs = append(optArgs, tt.opts...)
+		optArgs = append(optArgs, c.env.ImagePath)
+		optArgs = append(optArgs, tt.argv...)
+
+		c.env.RunSingularity(
+			t,
+			e2e.AsSubtest(tt.name),
+			e2e.WithPrivileges(true),
+			e2e.WithCommand("exec"),
+			e2e.WithArgs(optArgs...),
+			e2e.ExpectExit(tt.expectExit),
+		)
+
+	}
+}
+
+// testSecurityConfOwnership tests checks on config files ownerships
+func (c *ctx) testSecurityConfOwnership(t *testing.T) {
+	configFile := buildcfg.SINGULARITY_CONF_FILE
+
+	c.env.RunSingularity(
```

Comment:
In this case you could execute singularity command privileged, so you could remove `e2e.Privileged` calls in `PreRun` and `PostRun` because with `e2e.WithPrivileges(true)` both `PreRun` and `PostRun` are executed with privileges

Reply:
But I dont want the singularity command executed as root. Otherwise this test wont seem to work.

Reply:
Right, my bad